File: 351002/Alexeichik/MyStartRestApplication/src/main/java/com/github/Lexya06/startrestapp/controller/realization/NoticeController.java
package com.github.Lexya06.startrestapp.controller.realization;

import com.github.Lexya06.startrestapp.controller.abstraction.BaseController;
import com.github.Lexya06.startrestapp.model.dto.request.NoticeRequestTo;
import com.github.Lexya06.startrestapp.model.dto.response.NoticeResponseTo;
import com.github.Lexya06.startrestapp.model.entity.realization.Notice;
import com.github.Lexya06.startrestapp.service.abstraction.BaseEntityService;
import com.github.Lexya06.startrestapp.service.realization.NoticeService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("${server.api.base-path.v1}/notices")
@Validated
public class NoticeController extends BaseController<Notice, NoticeRequestTo, NoticeResponseTo> {
    NoticeService noticeService;
    @Autowired
    public NoticeController(NoticeService noticeService) {
        this.noticeService = noticeService;
    }
    @Override
    protected BaseEntityService<Notice, NoticeRequestTo, NoticeResponseTo> getBaseService() {
        return noticeService;
    }
}

File: 351003/brezgunov/distcomp/src/main/kotlin/com/distcomp/controller/NoticeController.kt
package com.distcomp.controller

import com.distcomp.dto.notice.NoticeRequestTo
import com.distcomp.dto.notice.NoticeResponseTo
import com.distcomp.service.NoticeService
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/{version}/notices")
class NoticeController(
    private val noticeService: NoticeService
) {
    @PostMapping(version = "1.0")
    @ResponseStatus(HttpStatus.CREATED)
    fun createNotice(@RequestBody noticeRequestTo: NoticeRequestTo): NoticeResponseTo {
        return noticeService.createNotice(noticeRequestTo)
    }

    @GetMapping("{id}")
    fun readNoticeById(@PathVariable("id") id: Long): NoticeResponseTo {
        return noticeService.readNoticeById(id)
    }

    @GetMapping
    fun findAll(): List<NoticeResponseTo> {
        return noticeService.readAll()
    }

    @PutMapping(path = ["/{id}", ""], version = "1.0")
    @ResponseStatus(HttpStatus.OK)
    fun updateNotice(
        @RequestBody noticeRequestTo: NoticeRequestTo,
        @PathVariable("id") id: Long?
    ): NoticeResponseTo {
        return noticeService.updateNotice(noticeRequestTo, id)
    }

    @DeleteMapping(path = ["/{id}"], version = "1.0")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    fun deleteNotice(@PathVariable("id") id: Long) {
        noticeService.removeNoticeById(id)
    }
}